src/main/java/com/salesforce/storm/spout/dynamic/DynamicSpout.java

```diff
@@ -420,9 +451,17 @@ public MetricsRecorder getMetricsRecorder() {
     /**
      * @return The virtual spout coordinator.
      */
-    SpoutCoordinator getCoordinator() {
+    SpoutCoordinator getSpoutCoordinator() {
```

Comment:
Interestingly the only time we call this is to get the number of spouts, `SpoutCoordinator.getTotalSpouts()`.

I'd like to see SpoutCoordinator NOT be public.  I think we should do one of two things...
1. Add a `DynamicSpout.getTotalVitualSpouts()` method that basically does what we need in the tests that call this method.
2. Add a `DynamicSpout.getVirtualSpouts()` which we can call `size()` on in the tests.  In the past I've debated adding this but the complexity with the monitor -> coordinator made me avoid doing so. With as simple as this is not it actually might be worth it.

If you want we can do that in a separate PR, just write a ticket up for it.

Reply:
So I could go either way on this, but I think in principle we're on the same page.

I feel like either SpoutHandlers deal w/ SpoutCoordinator to start/stop/whatever virtual spouts because it has a limited API (compared to DynamicSpout)

OR

You don't get access to SpoutCoordinator at all, and you're funneled thru DynamicSpout.

Reply:
Right now the SpoutHandlers deal directly with the DynamicSpout. I'm definitely not opposed to changing that, but for all intents and purposes DynamicSpout is our principal API surface area.

Reply:
sounds good